### PR TITLE
Fix published flag

### LINKED ISSUE #383: Add published flag column to all user-writable tables
VERY IMPORTANT: add a spec that tests if there are no tables which could be populated by user, either via the submission system, or via API, and that do not have the new published flag.

Default all current data to published, not private.

*OR* add the flag field to *ALL* tables (which makes the spec mentioned above not required). This might be more practical option.

### PR DESCRIPTION
**Pull request by nowakowski**
Fixes #383. As requested, I did not touch the existing implementation of `published?` in AR classes (requires further input from @wjurkowski).

### CODE REVIEW

**Comment by Nuanda** [comment]
@kammerer Is that ok with you?

**Comment by Nuanda** [comment]
Guys, let's wait with that one as well - I've just got more data from Wiktor. Will email you both.

**Comment by nowakowski** [comment]
Keeping abreast of the dynamic discussion going on in #383 :). On @Nuanda's request I also dropped the `publishable` flag in `submissions`.

**Comment by Nuanda** [comment]
Tested submissions manually and they seem to work ok. Please, take into consideration these three requests:
- [ ] separate adding timestamps to another migration
- [ ] the two PV->countries relations now have published without published_on - remove the published flag (you can still do that in the 20160317135309 migration and rename it afterwards)
- [ ] remove both new fields (published, published_on), and the validation, from all models that do not have user_id - I think it does not make sense to have e.g. private data with no ownership, right?

**Comment by Nuanda** [comment]
Why did your migrations removed so many `on_update: :cascade` callbacks from schema.rb? Are they important or not?

**Comment by nowakowski** [comment]
Hm; `schema.rb` is generated, not written. I'm not sure how exactly Rails generates this file but I've checked the model classes and there seems to be no reason to enforce cascading deletes on the DB level (there are no `dependent` conditions in AR classes). I believe the present form of `schema.rb` is correct.

File: app/services/submission/plant_population_finalizer.rb [hunk 1]
```diff
@@ -30,7 +30,9 @@ def create_new_plant_lines
         taxonomy_term_id: taxonomy_term.id,
         entered_by_whom: submission.user.full_name,
         date_entered: Date.today,
-        user: submission.user
+        user: submission.user,
+        published: true,
+        published_on: Date.today
```
Comment (Nuanda) [inline app/services/submission/plant_population_finalizer.rb]: Here you use Date.today yet in other places you set that column to Time.now. Which is correct?

Comment (nowakowski) [inline app/services/submission/plant_population_finalizer.rb]: You're right, it should be `Time.now`. Fixed.

**Comment by Nuanda** [comment]
@nowakowski Piotr, when you extend the correct code with new functions, please also extend corresponding specs do the new feature is being tested as well.

**Comment by nowakowski** [comment]
@Nuanda Acknowledged. Is there anything specific to be done in the context of this PR?

**Comment by Nuanda** [comment]
:+1:

**Pull request merged**